object/task.go

```diff
@@ -64,3 +64,37 @@ func (t *Task) Cancel(ctx context.Context) error {
 
 	return err
 }
+
+// SetTaskState sets task state and optionally sets results or fault, as appropriate for state.
+func (t *Task) SetTaskState(ctx context.Context, state types.TaskInfoState, result types.AnyType, fault *types.LocalizedMethodFault) error {
```

Comment:
Since this is a Task method, let's remove 'Task' from the name: `SetState` instead of `SetTaskState`

Reply:
You're so right.  Good call!